<div dir="ltr" style="text-align: left;" trbidi="on">
<div class="separator" style="clear: both; text-align: center;">
</div>
<div class="separator" style="clear: both; text-align: center;">
<a href="http://1.bp.blogspot.com/-Jtu_tWm8f_0/TklTreAI3mI/AAAAAAAAAIE/O_QXipu6fUg/s1600/lociNGS3.jpg" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;"><img border="0" height="200" src="http://1.bp.blogspot.com/-Jtu_tWm8f_0/TklTreAI3mI/AAAAAAAAAIE/O_QXipu6fUg/s200/lociNGS3.jpg" width="194" /></a></div>
<a href="http://4.bp.blogspot.com/-ATT7vOxuK-Y/Tkk0ND-STPI/AAAAAAAAAIA/Ntc-pcTbM-s/s1600/lociNGS.jpg" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;">
</a>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;"><span style="color: lime;"><span style="font-size: large;">lociNGS</span></span><span style="font-size: large;">: a simple database to display and reformat large multi-locus datasets&nbsp;</span></span><span style="font-size: large;">  </span><span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;"><span style="font-size: large;">&nbsp;</span></span>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">This program is the project I wrote for <span style="color: lime;"><a href="http://socghop.appspot.com/gsoc/homepage/google/gsoc2011">Google Summer Of Code 2011</a></span> under guidance from <span style="color: lime;">Jeremy Brown</span> and <span style="color: lime;">Brad Chapman</span> with <span style="color: lime;"><a href="http://informatics.nescent.org/wiki/Phyloinformatics_Summer_of_Code_2011">NESCent</a></span>. It is written in Python and uses <span style="color: lime;"><a href="http://www.mongodb.org/">MongoBD</a></span> for it's databasing functions. Please feel free to contact me with any issues.&nbsp;</span>  <span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">&nbsp;</span>
</div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">
</span></div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">lociNGS can be downloaded from GitHub <a href="https://github.com/SHird/lociNGS">here</a>.</span>
</div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">A copy of the README file is available with the program distribution and <a href="https://github.com/SHird/lociNGS/blob/master/README.txt">here</a>.&nbsp;</span>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">
</span></div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">
</span></div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">One final screenshot:</span>
</div>
</div>
<div class="separator" style="clear: both; text-align: center;">
<a href="http://3.bp.blogspot.com/-0cij1Blq0yE/TklUKXkOlBI/AAAAAAAAAII/x4Hz4cYdSzk/s1600/programWindows.jpg" imageanchor="1" style="clear: left; float: left; margin-bottom: 1em; margin-right: 1em;"><img border="0" height="400" src="http://3.bp.blogspot.com/-0cij1Blq0yE/TklUKXkOlBI/AAAAAAAAAII/x4Hz4cYdSzk/s640/programWindows.jpg" width="640" /></a></div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">
</span>
</div>
<span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">Sarah Hird&nbsp;</span> <span style="color: lime;"><span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">&nbsp;</span></span>
</div>
<span style="color: lime;"><span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;">shir</span></span><span style="font-family: &quot;Courier New&quot;, &quot;Courier&quot;, monospace;"><span style="color: lime;">d1@tigers.lsu.edu</span></span></div>


<div dir="ltr" style="text-align: left;" trbidi="on">
   
<div class="MsoNormal">
lociNGS [v.1] README<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Copyright (C)<span style="mso-spacerun: yes;">&nbsp; </span>2011<span style="mso-spacerun: yes;">&nbsp; </span>Sarah M. Hird </div>
<div class="MsoNormal">
Permission is granted to copy, distribute and/or modify this document under the terms of the GNU Free Documentation License, Version 1.3 or any later version published by the Free Software Foundation; with no Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts. A copy of the license is included in the document entitled "GNU Free Documentation License".</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
LOCINGS: a simple database for reformatting and displaying multi-locus datasets</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
-----------------------------------------------------------------</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1. INSTALLATION</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.1. MongoDB</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.1.1. Installation of MongoDB</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.1.2. Correctly shutting down MongoDB</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.1.3. How to tell if your computer is 32- or 64- bit</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.2. lociNGS Easy Install</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3. More info</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.1. Python installation</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.2. NumPy</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.3. Biopython</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.4. Pymongo</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.5. Pysam</div>
<div class="MsoNormal">
<span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.3.6. seq_lite.py</div>
<div class="MsoNormal">
2. TO RUN lociNGS</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.1. Short answer</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.2. Contents of the folder</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.3. Starting MongoDB</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.4. Starting lociNGS</div>
<div class="MsoNormal">
3. INPUT FORMATS</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3.1 Import types</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3.2 Locus Names **IMPORTANT**</div>
<div class="MsoNormal">
4. IMPORTING DATA</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4.1. Loci/fasta file(s)</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4.2. SAM/BAM NGS data</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4.3. Demographic data </div>
<div class="MsoNormal">
5. THE lociNGS INTERFACE</div>
<div class="MsoNormal">
6. EXPORTING DATA</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>6.1. NEXUS format</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>6.2. IMa2 format</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>6.3. Migrate format</div>
<div class="MsoNormal">
7. TEST DATA</div>
<div class="MsoNormal">
8. WHAT IF...</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8.1. ...lociNGS won't start?</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8.2. ...locus screen coverage buttons all display "0"?</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8.3. ...summary screen shows no data?</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8.4. Problems with IMa2 output.</div>
<div class="MsoNormal">
9. CONTACT</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
----------------------------------------------------------------</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1. INSTALLATION</div>
<div class="MsoNormal">
The program is dependent on several other pieces of software. lociNGS was written on MacOSX. lociNGS is available for download at https://github.com/SHird/lociNGS.<span style="mso-spacerun: yes;">&nbsp; </span>You will have to install MongoDB separately from the rest of the package.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.1 MongoDB</div>
<div class="MsoNormal">
1.1.1 Installation of MDB</div>
<div class="MsoNormal">
Go to www.mongodb.org/downloads; </div>
<div class="MsoNormal">
Download correct version (if you don't know if your machine is 32 or 64 bit, see section 1.1.3.);</div>
<div class="MsoNormal">
Double click the downloaded file – this should unpack it into a folder called something like "mongodb-osx-x86_64-1.8.2" (I would rename this folder "MongoDB" but you don't have to);</div>
<div class="MsoNormal">
Move the MongoDB folder to Applications (or wherever you want to keep it. ***You will need mongod running every time you use lociNGS, so you should remember where you put the MongoDB folder.);</div>
<div class="MsoNormal">
Make the directory that stores the data by opening a terminal and typing "mkdir –p /data/db" (without the quotations)</div>
<div class="MsoNormal">
Go to the mongoDB folder, then the bin folder. </div>
<div class="MsoNormal">
Double click mongod. This should open a screen with something like this:</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Last login: Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 08:12:53 on ttys001</div>
<div class="MsoNormal">
/Applications/mongodb-osx-x86_64-1.8.1/bin/mongod ; exit;</div>
<div class="MsoNormal">
HappyPappy:~ shird$ /Applications/mongodb-osx-x86_64-1.8.1/bin/mongod ; exit;</div>
<div class="MsoNormal">
/Applications/mongodb-osx-x86_64-1.8.1/bin/mongod --help for help and startup options</div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [initandlisten] MongoDB starting : pid=11886 port=27017 dbpath=/data/db/ 64-bit </div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [initandlisten] db version v1.8.1, pdfile version 4.5</div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [initandlisten] git version: a429cd4f535b2499cc4130b06ff7c26f41c00f04</div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [initandlisten] build sys info: Darwin erh2.10gen.cc 9.6.0 Darwin Kernel Version 9.6.0: Mon Nov 24 17:37:00 PST 2008; root:xnu-1228.9.59~1/RELEASE_I386 i386 BOOST_LIB_VERSION=1_40</div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [initandlisten] waiting for connections on port 27017</div>
<div class="MsoNormal">
Wed Aug<span style="mso-spacerun: yes;">&nbsp; </span>3 09:13:53 [websvr] web admin interface listening on port 28017</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Leave this window open as you run lociNGS.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.1.2. Correctly shutting down MongoDB</div>
<div class="MsoNormal">
When you are finished with lociNGS and MongoDB, you must shut down MongoDB correctly. Click on the terminal window then press Control + C (the control button and the "c" button at once). If you don't do this, you'll have to find a file and delete it before you can get MongoDB running again – spotlight the file "mongod.lock" then drag it to the trash.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.1.3. How to tell if your computer is 32- or 64-bit (from support.apple.com/kb/ht3696)</div>
<div class="MsoNormal">
1. Choose About This Mac from the Apple (<span style="font-family: Monaco; mso-bidi-font-family: Monaco;"></span>) menu in the upper-left menu bar, then click More Info.</div>
<div class="MsoNormal">
2. Open the Hardware section.</div>
<div class="MsoNormal">
3. Locate the Processor Name [under the Hardware Overview].</div>
<div class="MsoNormal">
4. Compare your Processor Name to information below to determine whether your Mac has a 32-bit or 64-bit processor.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Processor Name<span style="mso-tab-count: 3;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>32- or 64-bit</div>
<div class="MsoNormal">
Intel Core Solo<span style="mso-tab-count: 4;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>32 bit</div>
<div class="MsoNormal">
Intel Core Duo <span style="mso-tab-count: 4;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>32 bit</div>
<div class="MsoNormal">
Intel Core 2 Duo<span style="mso-spacerun: yes;">&nbsp; </span><span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span style="mso-spacerun: yes;">&nbsp;&nbsp;</span><span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Intel Quad-Core Xeon<span style="mso-spacerun: yes;">&nbsp;&nbsp; </span><span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Dual-Core Intel Xeon<span style="mso-spacerun: yes;">&nbsp;&nbsp; </span><span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Quad-Core Intel Xeon<span style="mso-spacerun: yes;">&nbsp;&nbsp; </span><span style="mso-tab-count: 2;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Core i3<span style="mso-tab-count: 6;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Core i5<span style="mso-tab-count: 6;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">
Core i7<span style="mso-tab-count: 6;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>64 bit</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.2. lociNGS Easy Install</div>
<div class="MsoNormal">
download the lociNGS package from GitHub. You may either go to the website [ https://github.com/SHird/lociNGS ] and click the "Downloads" button (then choose the .tar.gz option). When the package has downloaded, double click the download and move the folder to the Applications folder (or wherever you'd like it to be).</div>
<div class="MsoNormal">
Alternatively, if you have git on your machine, you can clone the directory by typing "git clone git@github.com:SHird/lociNGS.git /Applications/lociNGS". (This will install lociNGS into the Applications folder on MacOSX. You may move it.)</div>
<div class="MsoNormal">
Open a terminal window, cd into the lociNGS folder and type "python setup.py install". This should install all the components in the appropriate places. If you see some output to the terminal screen and the last line is "Finished processing dependencies for lociNGS==1.0", you're all set.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3 More information</div>
<div class="MsoNormal">
1.3.1 Python installation</div>
<div class="MsoNormal">
MacOSX should come with a version of Python already installed. If you're not sure Python is on your machine:</div>
<div class="MsoNormal">
In an open terminal, type "python". This should display something like "Python 2.7.1 (r271:86882M, Nov 30 2010, 10:35:34)" plus a couple of other lines of text followed by "&gt;&gt;&gt;". Type "exit()".</div>
<div class="MsoNormal">
If the above did not happen, go to www.python.org/getit/ for Python installation.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3.2 NumPy<span style="mso-spacerun: yes;">&nbsp; </span>- http://sourceforge.net/projects/numpy/files/ (they have an automated installer, download the file, double click, follow instructions)</div>
<div class="MsoNormal">
For more information: http://www.scipy.org/Download</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3.3 Biopython - http://biopython.org/wiki/Download; download, double click, move the folder into the home directory, open terminal window, cd into folder, type three commands: [1] "python setup.py build", [2] "python setup.py test", [3] "sudo python setup.py install"</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3.4 PyMongo - https://github.com/mongodb/mongo-python-driver; click "Downloads" button (upper-ish right of screen); click "1.11";<span style="mso-spacerun: yes;">&nbsp; </span>once it downloads, rename the folder "pymongo", put the folder in the home directory, cd into it, and type "python setup.py install"</div>
<div class="MsoNormal">
For more information, other installation options: http://api.mongodb.org/python/current/installation.html</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3.5 pysam - http://code.google.com/p/pysam/downloads/list; download file, double-click, move folder into Applications folder (or where you'd like to keep it), cd into folder and type two commands: [1] "python setup.py build", [2] "python setup.py install"</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
1.3.6 seqlite_mod.py – this module is included with the distribution and you should have to do nothing to compile or install it.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
2. TO RUN lociNGS:</div>
<div class="MsoNormal">
2.1. Short answer: find and double click "mongod"; type "run_lociNGS.py" in a terminal window</div>
<div class="MsoNormal">
2.2. Contents of the folder</div>
<div class="MsoNormal">
I think it's easiest to keep everything in one place, but you don't necessarily have to. You need the following components in the lociNGS folder in order to run it:</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>1.<span style="mso-spacerun: yes;">&nbsp; </span>locings folder</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>2.<span style="mso-spacerun: yes;">&nbsp; </span>scripts folder</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>3.<span style="mso-spacerun: yes;">&nbsp; </span>setup.py</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>4.<span style="mso-spacerun: yes;">&nbsp; </span>README.txt</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>5.<span style="mso-spacerun: yes;">&nbsp; </span>locus folder of fasta files*</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>6.<span style="mso-spacerun: yes;">&nbsp; </span>BAM and BAI folder*</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>7.<span style="mso-spacerun: yes;">&nbsp; </span>Tab-delimited text file of demographic data* </div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>8.<span style="mso-spacerun: yes;">&nbsp; </span>IMa2 input file (if formatting for IMa2 or Migrate, format described below)</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
*These folders/files don't need to be in the lociNGS folder.</div>
<div class="MsoNormal">
After the program is run for the first time, an additional four files will appear with the extension .pyc (for each of the python scripts above). These can be removed but are used to speed up the running of python scripts.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
2.3. Starting MongoDB</div>
<div class="MsoNormal">
MongoDB must be running before lociNGS is started. To do this, you can find "mongod" with the Finder and double click it; or go to the MongoDB/bin folder and double click "mongod"; or, from the terminal, you can cd into the MongoDB folder, then cd into "bin" and type "./mongod". Leave the window open and when you are finished with lociNGS YOU MUST EXIT MONGOD BY PRESSING CONTROL+C (TOGETHER). If you forget to do this, the mongod process will be improperly shut down and not start the next time you need it. If that happens, see Section 1.1.2. above.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
2.4 Starting lociNGS</div>
<div class="MsoNormal">
To run the program, open a terminal window and type "run_lociNGS.py". This should open a new window with these basic instructions: "Please enter the data in the order listed in the Import Menu. Once data has been loaded via the Import Menu, press ‘Display the data'." Don't close this window – it will close the program.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
3. INPUT FORMATS</div>
<div class="MsoNormal">
3.1 Import types</div>
<div class="MsoNormal">
lociNGS uses three imports. </div>
<div class="MsoNormal">
[1] The first is a folder of files that contain loci in fasta format. These files should have ".fasta" as their final extension. The folder may be located anywhere.</div>
<div class="MsoNormal">
[2] The second import is a folder of indexed bam files from a short read aligner. Each bam (and corresponding .bai) file should correspond to an individual in the dataset. I'm working on getting sam format to work too, but for right now, indexed bam works best.</div>
<div class="MsoNormal">
[3] The third is a tab-delimited text file that contains demographic data for the individuals in the dataset. There must be at least two columns, labeled "Individual" and "Population", which contain information on the name of the individual (as it appears in locus files and BAM file names) and which population the individual came from (these can be numbers or letters). The file may contain as many columns as you'd like, they will appear on the "Summary Screen" of the program.</div>
<div class="MsoNormal">
*lociNGS will run and reformat loci if just [1] loci and [3] demographic data are entered. </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
3.2 Locus Names **IMPORTANT**</div>
<div class="MsoNormal">
The names of the files from import [1] that correspond to loci need to match with the locus names in the BAM files, import [2]. Basically, if your loci are called "locus1", "locus2" and "locus3", the fasta files need to be called "locus1.fasta", "locus2.fasta" and "locus3.fasta" and the loci in your BAM file need to be "locus1", "locus2" and "locus3". There are a few exceptions - the locus file name comes from anything before the first "." in the locus file name - so the files could be called "locus1.aln.fasta" or "locus1.080911.fasta" as long as the prefix before the first "." matches the BAM loci. Also, the BAM loci may have prefixes, but this time, due to common GenBank annotation, you may add things to the BAM loci names with a "|" as a separator. So BAM loci may be called something like "gi|323|testdata|locus1" as long as the last piece of the header matches the fasta loci names.</div>
<div class="MsoNormal">
Any combination of the below files will work in lociNGS. Please feel free to contact me if you are unsure if your locus names are causing the program difficulty.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
FASTA FILE - must contain the locus name before the first period and have ".fasta" as final extension</div>
<div class="MsoNormal">
locus1.fasta</div>
<div class="MsoNormal">
locus1.aln.fasta</div>
<div class="MsoNormal">
locus1.test.data.fasta</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
BAM LOCUS NAME - must contain the locus name after the final | or as the only text</div>
<div class="MsoNormal">
locus1</div>
<div class="MsoNormal">
fake|any text you want|locus1</div>
<div class="MsoNormal">
gi|323|fakeData|08august2011|locus1</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
4. IMPORTING DATA</div>
<div class="MsoNormal">
4.1 Loci/fasta file(s)</div>
<div class="MsoNormal">
Step 1 in the import menu will open a window where you should find and select the locus folder. After successful import, the lociNGS screen will tell you. The terminal window will print data as the files are read. It should be a lot of text that looks something like: </div>
<div class="MsoNormal">
"Got this file:<span style="mso-spacerun: yes;">&nbsp; </span>/Users/shird/Desktop/juncoLoci/JUNCOmatic_63_aln.fasta</div>
<div class="MsoNormal">
locusFasta =<span style="mso-spacerun: yes;">&nbsp; </span>JUNCOmatic_63_aln.fasta ; individuals {'J12': 0, 'J09': 0, 'J18': 0, 'J19': 0, 'J01': 0, 'J17': 0, 'J03': 0, 'J11': 0, 'J05': 0, 'J04': 0, 'J10': 0, 'J06': 0} ; indInFasta ['J12', 'J09', 'J18', 'J19', 'J01', 'J17', 'J03', 'J11', 'J05', 'J04', 'J10', 'J06'] ; SNPs =<span style="mso-spacerun: yes;">&nbsp; </span>5 ; number alleles =<span style="mso-spacerun: yes;">&nbsp; </span>24 ; length =<span style="mso-spacerun: yes;">&nbsp; </span>284 ; path =<span style="mso-spacerun: yes;">&nbsp; </span>/Users/shird/Desktop/juncoLoci/JUNCOmatic_63_aln.fasta"</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
4.2 SAM/BAM NGS data</div>
<div class="MsoNormal">
Step 2 will import the net-gen alignments - you should find and select the indexed bam folder. lociNGS will update as the import is finished. The terminal window will print data as the files are read. It will look something like this for each file: </div>
<div class="MsoNormal">
"Got this folder: /Users/shird/Documents/Dropbox/juncoBam</div>
<div class="MsoNormal">
Got this file:<span style="mso-spacerun: yes;">&nbsp; </span>/Users/shird/Documents/Dropbox/juncoBam/J01.sorted.bam</div>
<div class="MsoNormal">
730</div>
<div class="MsoNormal">
individuals.J01"</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
4.1.3 Demographic data</div>
<div class="MsoNormal">
The final step imports the demographic data in a tab delimited file - find and select the tab delimited demographic data file. Again, lociNGS will tell you when the import has been successful and the terminal will print data for each individual as the files are read:</div>
<div class="MsoNormal">
"{'Longitude': '-109.876', 'Individual': 'J01', 'Location': 'NoPlace, TX', 'Latitude': '45.678', 'Species': 'Junco hyemalis', 'Population': 'POP1'}</div>
<div class="MsoNormal">
POP1"</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
5. THE lociNGS INTERFACE</div>
<div class="MsoNormal">
lociNGS has three screens meant to show you how much data is associated with each individual in your dataset. The first screen will display text updates as the program completes functions. The second is a summary screen where each individual is a row and the demographic data are the columns. There is also a "numLoci" column that displays the number of loci called for that particular individual. If you click one of these numLoci buttons, the third screen appears that displays the specific loci. On this screen there are five columns:</div>
<div class="MsoNormal">
Locus Name = the locus file/locus name</div>
<div class="MsoNormal">
Length = length of the locus</div>
<div class="MsoNormal">
Coverage_This_Ind = how many raw reads from this individual aligned to the locus. If this button is pressed, a fasta file is generated that contains these reads. This file will be printed to the directory that contains the lociNGS scripts. </div>
<div class="MsoNormal">
Number_Inds = how many individuals are present in the locus (fasta) file</div>
<div class="MsoNormal">
Coverage_Total = how many raw reads from any individual aligned to the locus. If this button is pressed, a fasta file that contains the reads is generated and printed to the lociNGS directory.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
6. EXPORTING DATA</div>
<div class="MsoNormal">
You may export data in three formats (NEXUS, IMa2 and MIGRATE) and you may select either specific individuals from your dataset you'd like to include in your files OR you may select a set of populations. If you select populations, lociNGS will scan the locus files for any locus that contains at least one individual from each of the selected populations and reformat these loci.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
6.1 NEXUS format</div>
<div class="MsoNormal">
NEXUS format requires aligned data. Files will be printed with a ".nex" extension.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
6.2 IMa2 format</div>
<div class="MsoNormal">
IMa2 requires an additional input file to describe some of the dataset specific parameters. I've included an example file. Please see IMa2 manual (available at http://lifesci.rutgers.edu/~heylab/ProgramsandData/Programs/IMa2/Using_IMa2_10_13_10.pdf) for more information on the parameters.</div>
<div class="MsoNormal">
The IMa2 extra parameters file must be named "IMa2InputFile.txt" and should follow the format of the included IMa2InputFile format – just type the values you need after the colon.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Header Line: Type whatever you want here that will identify your IMa run</div>
<div class="MsoNormal">
Population Tree: ((0,1):5,(2,3):4):6</div>
<div class="MsoNormal">
Inheritance Scalar: 1</div>
<div class="MsoNormal">
Mutation Model: H</div>
<div class="MsoNormal">
Mutation Rate (optional): 0.000008</div>
<div class="MsoNormal">
Mutation Rate Range Lower (optional): 0.000007</div>
<div class="MsoNormal">
Mutation Rate Range Upper (optional): 0.000009</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
If you aren't using the Mutation Rate Parameters, delete them from the file. If you are, please note that the mutation rate you enter needs to be the substitutions / site mutation rate; lociNGS will multiply this rate by the length of each locus (# of sites) to give a per base mutation rate (which is what the program requires). Also note that the Population Tree is required and is a very specific format, so please check with the IMa2 documentation for how to write it.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Also note, the same inheritance scalar and mutation model will be printed for every locus. If you have different types of data (autosome, mtDNA, X- or Y-linked data), you might want to put the similar types together in a folder and load them independently into lociNGS, and export with the correct inheritance scalar in the IMa2InputFile. You can then concatenate the files.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
6.3 Migrate format</div>
<div class="MsoNormal">
Migrate format requires that an IMa2 additional file be in the folder, but doesn't use the information – so if you just need Migrate output, leave the example IMa2InputFile.txt in the folder as is.</div>
<div class="MsoNormal">
<span style="mso-tab-count: 1;">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></div>
<div class="MsoNormal">
7. TEST DATA</div>
<div class="MsoNormal">
I've included a very small test dataset, containing four individuals and five loci. </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
8. WHAT IF...? </div>
<div class="MsoNormal">
I've attempted compiling a list of potential problems - email me (please) if you encounter something not on this list so I can add it for other users.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
8.1. ...lociNGS won't start? </div>
<div class="MsoNormal">
If this ("pymongo.errors.AutoReconnect: could not find master/primary") is the last line of output printed to the terminal window, it means that mongod is not running. In the MongoDB/bin folder, double click on "mongod" then try starting lociNGS again.</div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
8.2. ...locus screen coverage buttons all say "0"?</div>
<div class="MsoNormal">
This could be due to the name of the loci in the SAM/BAM files not corresponding correctly to the name of the fasta locus files. Double check by referring to section 3.2 above. </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
8.3. ...summary screen shows no data?</div>
<div class="MsoNormal">
This could be due to the names of individuals not corresponding correctly between individuals in the fasta files and the demographic data text file. The demographic table needs to have a minimum of two columns, "Individual" and "Population". Make sure the names of the indivdiuals are the same as the fasta files. </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
8.4. Problems with IMa2.</div>
<div class="MsoNormal">
The IMa2InputFile.txt needs to look almost exactly like the one I've provided. Please let me know if you have specific problems with any of the output formats. </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
9. CONTACT</div>
<div class="MsoNormal">
Please feel free to contact me about any issues you're having with lociNGS or the dependent software. I'd be more than happy to do what I can – </div>
<div class="MsoNormal">

</div>
<div class="MsoNormal">
Sarah Hird</div>
<div class="MsoNormal">
shird1@tigers.lsu.edu</div>

</div>
